[Arya Nagatama Giat] iP

aryagiat opened the pull request:
# Duke x BORAT

> Very nice! - Borat Sagdiyev

### Forget your tasks no more with _**Duke x BORAT**_. Now you can:
- [ ] Add *todo* tasks
- [ ] Add *deadlines*
- [ ] Add *events*
- [ ] Mark them as done
- [ ] Delete tasks
- [ ] Find tasks

### Why _**Duke x BORAT**_?
- **SIMPLE!** 💎
- **Incredibly** Fast! ⚡
- Data ~can be leaked~ is **SAFE** with you and only you 😏
- **EASY** ✏

### How to use?
1. Download it from [here](https://github.com/aryagiat/ip/releases/tag/v0.1) (**Definitely** safe)
2. Open it
3. do `help` to see all commands
4. add your first todo task by doing:
```
todo watch Borat! Cultural Learnings of America for Make Benefit Glorious Nation of Kazakhstan
```
<hr>

### Help develop:
Here is the entry point: go to `Duke.java`:

``` java
/**
 * Entry point of the Duke program.
 * @param args Command line arguments.
 */
public static void main(String[] args) {
    new Duke("./data", "duke.txt").run();
}
```

Commit:
Add Gradle support

aryagiat committed:
Feat: Greet, Echo, Exit

aryagiat committed:
Feat: Add, List

aryagiat committed:
Feat: Mark as done

aryagiat committed:
Feat: Todos, Events, Deadlines

aryagiat committed:
Feat: Automated Text UI Testing

aryagiat committed:
Feat: add exception handling

aryagiat committed:
Feat: Delete functionality

aryagiat committed:
Feat: enums

aryagiat committed:
Feat: read from saved file and parse it

aryagiat committed:
Feat: Update the saved data on updates

aryagiat committed:
Chore: clean saved data

aryagiat committed:
Chore: merge with changes from main

aryagiat committed:
Feat: Add a help function to show all available commands

aryagiat committed:
Chore: change date from string to java Date

aryagiat committed:
Feat: add date parser

aryagiat committed:
Chore: encapsulate DukeException messages to an enum

aryagiat committed:
Add Constants class to hold enums and Ui to deal with ui stuff

aryagiat committed:
Add Storage class

aryagiat committed:
Add Parser class

aryagiat committed:
add comments

aryagiat committed:
Add all to duke package

aryagiat committed:
Add jUnit Tests

aryagiat committed:
Add javadocs

aryagiat committed:
Fix tab spacing

aryagiat committed:
Merge branch 'branch-A-CodingStandard'

aryagiat committed:
Add feature to find tasks based on a given keyword

aryagiat committed:
Merge branch 'branch-Level-9'

aryagiat committed:
Fix: comply to java coding standards

beefham commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,153 @@
+    private final Parser parser;
+
+    /** The list of tasks */
+    private TaskList list;
```
Perhaps "taskList" or "tasks" may be a better name for this variable.

beefham commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,40 @@
+/**
+ * Exceptions for the Duke app.
+ */
+public class DukeException extends Exception{
```
Very small nitpick, there is a missing space between "Exception" and "{"

beefham commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,316 @@
+
+        case DEADLINE:
+            if (inputs.length < 2) {
+                throw new DukeException(DukeException.Errors.MISSING_DESCRIPTION.toString() + " (example: 'deadline watch Borat /by 2021-08-21 18:00')");
```
I like the use of an example string to guide the user to a correct format for each command, but perhaps this line should be broken up because it is too long (over 120 characters)

beefham commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+/**
+ * A Task of type Deadline.
+ */
+public class Deadline extends Task{
```
Minor nitpick, there is a missing space between "Task" and "{"

stephanie-csy commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,316 @@
+            if (
+                    (splitTime[0].length() == 2 || splitTime[0].length() == 1) &&
+                            (splitTime[1].length() == 2)
+            ) {
```
Not sure if this is the correct way to split an `if` condition into different lines. Maybe 

```javascript
 if ((splitTime[0].length() == 2 || splitTime[0].length() == 1) 
    && (splitTime[1].length() == 2)) {
```
would be better? Applies to several instances of `if` below as well.

stephanie-csy commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,46 @@
+     * Returns the string representation of a task.
+     * @return The description of the task.
+     */
+    public String toString() {
```
Nitpicking but can consider adding an `@Override` above the method as it overrides the toString() method in Object.

stephanie-csy reviewed (commented):
Very neat and easily readable code, follows coding standards closely generally except for some tiny spacing/indentation errors here and there. Methods and variables are named appropriately. Really like that you handled all kinds of exceptions and gave the user information on what exactly the error is and what they should do instead. Also really like the Borat references hahaha good job!

aryagiat committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

aryagiat committed:
Update main class name

aryagiat committed:
Fix to match checkstyle

aryagiat committed:
Add missing space

aryagiat committed:
Add javaFx

aryagiat committed:
Add GUI

aryagiat committed:
Update main entry point for gradle build

aryagiat committed:
Add Java Assert

Add Java Assert to all class methods which takes in a nullable argument.
If the parameter is a null, it will raise an unchecked error.
This assertion is to make sure that no arguments are null.

aryagiat committed:
Remove extra empty lines for better spacing

aryagiat committed:
Add javadocs to the public method

aryagiat committed:
Change variable name to be more meaningful

aryagiat committed:
Replace variables with a more meaningful name

aryagiat committed:
ParseInput now returns a hashmap instead of a list.

Previously, parseInput returns a list of String from the parsed input.
This method works but is difficult to read and modify.
Now a HashMap is used to read the parsed input.

aryagiat committed:
Merge pull request #1 from aryagiat/branch-A-Assertions

Add Java Assert

aryagiat committed:
Merge with latest changes from master branch

aryagiat committed:
Merge pull request #2 from aryagiat/branch-A-CodeQuality

Improved Code Quality

aryagiat committed:
Add GitHub Actions to set up Continuous Integration (CI)

aryagiat committed:
Add GitHub Actions to set up Continuous Integration (CI)

aryagiat committed:
Merge pull request #3 from aryagiat/branch-A-CI

Add CI

aryagiat committed:
Add a feature to accept a more natural date.

Previously, the accepted dates are 'today', 'tomorrow', and a
'yyyy-mm-dd'.
Now, commands such as 'mon' or 'monday' can be accepted.
All other days are also accepted.
When a day is written, it will pick the next closest date for that day.

aryagiat committed:
Merge branch 'master' into branch-C-NaturalDates

# Conflicts:
#	src/main/java/duke/Parser.java

aryagiat committed:
Merge with latest changes from master branch

aryagiat committed:
Merge pull request #4 from aryagiat/branch-C-NaturalDates

Add a feature to accept a more natural date.

aryagiat committed:
Abstract commands into their own class

aryagiat committed:
Abstract commands into their own class

aryagiat committed:
Abstract out commands into their own class and execute them at appropriate places

aryagiat committed:
Add comments to public classes

aryagiat committed:
Merge pull request #6 from aryagiat/branch-A-CodingStandard

Improve coding quality

aryagiat committed:
Adjust minimum height for the dialog box

aryagiat committed:
Add colors to the GUI app

aryagiat committed:
Enlarge font size for the GUI

aryagiat committed:
Change button background color

aryagiat committed:
Change the background color of the duke bot dialog box

aryagiat committed:
Add icon for the app

aryagiat committed:
Fix checkstyle

aryagiat committed:
Fix style

aryagiat committed:
Fix Style

aryagiat committed:
Update checkstyle and test case

aryagiat committed:
Merge pull request #7 from aryagiat/branch-A-BetterGui

Better GUI

aryagiat committed:
Add greeting message when app starts

aryagiat committed:
Add user guide to the documentation

aryagiat committed:
Fix typo in response message

aryagiat committed:
Merge pull request #8 from aryagiat/branch-A-UserGuide

Add User Guide

aryagiat committed:
Set theme jekyll-theme-leap-day

aryagiat committed:
Update README.md

aryagiat committed:
Rename the jar release file

aryagiat committed:
Update app download link to the latest link
